### DISCUSSION TITLE
Generating client queries from nested types only creates top level type

### DISCUSSION THREAD

**Discussion by anderssandven** [Netflix/dgs-framework#370] [Q&A]
I have a schema like this:
```
type Mutation {
	one: One
}

type One {
	two(three: Three!): Two
}

type Two {
	test: String
}

input Three {
	test3: String
}
```

When doing code generation I expect to get multiple queries generated. The queries generated are:
OneGraphQLQuery.java  
OneProjectionRoot.java  
OneTwoProjection.java

I expect to get a OneTwoGraphlQuery aswell because I cannot supply any arguments through OneGraphqlQuery. 

Am I misunderstanding how this works?

**Comment by nick-stephen** [comment]
I have been having a similar problem with this limitation, and have filed a feature request with details:

https://github.com/Netflix/dgs-framework/issues/547
